extensions/nodes/nodes.go

```diff
@@ -0,0 +1,92 @@
+package nodes
+
+import (
+	"errors"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/kubeapi/secrets"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	initNodeLabelKey      = "rke.cattle.io/init-node"
+	local                 = "local"
+	machinePlanSecretType = "rke.cattle.io/machine-plan"
+	machineNameSteveLabel = "rke.cattle.io/machine-name"
+)
+
+// GetInitMachine accepts a client and clusterID and returns the "init node" machine
+// object for rke2/k3s clusters
+func GetInitMachine(client *rancher.Client, clusterID string) (*v1.SteveAPIObject, error) {
+	logrus.Info("Retrieving list of secrets...")
+	secretList, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	machinePlans := []corev1.Secret{}
+
+	logrus.Info("Identifying machine-plan secrets...")
+	for _, secret := range secretList.Items {
+		if secret.Type == machinePlanSecretType {
+			logrus.Infof("Found machine-plan secret: %s", secret.Name)
+			machinePlans = append(machinePlans, secret)
+		}
+	}
+
+	initNodeMachineName := ""
+
+	logrus.Info("Identifying init node machine plan...")
+	for _, machinePlan := range machinePlans {
+		v, ok := machinePlan.ObjectMeta.Labels[initNodeLabelKey]
+		if ok && v == "true" {
```

Comment:
Josh, could you please check this: https://github.com/rancher/shepherd/pull/64#discussion_r1525497184. You can find more info on offline PR guidelines docs (15th)

Reply:
Hey Ugur,

I reviewed the comment, and when implementing it, I was getting a type error where bool cannot be type string, so I hardcoded this to "true" as it should never change. I felt this resolved the initial concerns with using the const and is a simple and clean resolution.

Do you have any concerns with this approach? 🙂